app/uk/gov/hmrc/claimvatenrolmentfrontend/connectors/AllocateEnrolmentConnector.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.claimvatenrolmentfrontend.connectors
+
+import play.api.libs.json.{JsObject, Json, Writes}
+import uk.gov.hmrc.claimvatenrolmentfrontend.config.AppConfig
+import uk.gov.hmrc.claimvatenrolmentfrontend.models.AllocateEnrolmentResponse
+import uk.gov.hmrc.claimvatenrolmentfrontend.models.AllocateEnrolmentResponseHttpParser.AllocateEnrolmentResponseReads
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class AllocateEnrolmentConnector @Inject()(http: HttpClient,
+                                           appConfig: AppConfig
+                                          )(implicit ec: ExecutionContext) {
+
+  def allocateEnrolment(groupId: String,
+                        credentialId: String,
+                        vatNumber: String)(implicit hc: HeaderCarrier): Future[AllocateEnrolmentResponse] = {
+    val enrolmentKey = s"HMRC-MTD-VAT~VRN~$vatNumber"
+
+    val requestBody = Json.obj(
+      "userId" -> credentialId,
+      "type" -> "principal",
+      "action" -> "enrolAndActivate"
+    )
+
+    http.POST[JsObject, AllocateEnrolmentResponse](
+      url = appConfig.allocateEnrolmentUrl(groupId, enrolmentKey),
+      body = requestBody
+    )(implicitly[Writes[JsObject]],
+      AllocateEnrolmentResponseReads,
+      hc,
+      ec)
+  }
+}
```

Comment:
This needs to be the ES8 call with known facts to verify that the user has entered them correctly. We're not doing a separate verification like on VAT sign up here.